R/metapredict_merge.R

```diff
@@ -32,33 +45,40 @@ makeMatchSampleMapping = function(metadata, subStudyNames, matchSampleColname) {
 #'
 #' @export
 mergeMatchStudyData = function(ematAtomicList, studyMetadataAtomic, matchStudyColname, sampleMetadataAtomic,
-                               matchSampleColname, mergeFunc=function(x) mean(x, na.rm=TRUE)) {
+                               matchSampleColname, mergeFunc = function(x) mean(x, na.rm = TRUE)) {
   ematList = list()
   sampleMetadataList = list()
+  sampleMetadataAtomicDT = data.table(sampleMetadataAtomic)
 
   for (matchStudyName in unique(studyMetadataAtomic[[matchStudyColname]])) {
-    if (sum(studyMetadataAtomic[[matchStudyColname]]==matchStudyName)==1) {
+    if (sum(studyMetadataAtomic[[matchStudyColname]] == matchStudyName) == 1) {
       ematList[[matchStudyName]] = ematAtomicList[[matchStudyName]]
-      sampleMetadataList[[matchStudyName]] = dplyr::filter(sampleMetadataAtomic, study==matchStudyName)
+      # sampleMetadataList[[matchStudyName]] = dplyr::filter(sampleMetadataAtomic, study == matchStudyName)
+      sampleMetadataList[[matchStudyName]] = sampleMetadataAtomicDT[study == matchStudyName,]
 
-    } else if (sum(studyMetadataAtomic[[matchStudyColname]]==matchStudyName)>1) {
-      atomicStudyNames = studyMetadataAtomic$study[studyMetadataAtomic[[matchStudyColname]]==matchStudyName]
+    } else if (sum(studyMetadataAtomic[[matchStudyColname]] == matchStudyName) > 1) {
+      atomicStudyNames = studyMetadataAtomic$study[studyMetadataAtomic[[matchStudyColname]] == matchStudyName]
       edfListNow = list()
       for (atomicStudyName in atomicStudyNames) {
         edf = data.frame(rownames(ematAtomicList[[atomicStudyName]]), ematAtomicList[[atomicStudyName]])
         rownames(edf) = NULL
-        matchNames = tibble::tibble(sample = colnames(edf)[2:ncol(edf)]) %>%
-          dplyr::inner_join(sampleMetadataAtomic, by='sample') %>%
-          .[[matchSampleColname]]
+        # matchNames = tibble::tibble(sample = colnames(edf)[2:ncol(edf)]) %>%
+        #   dplyr::inner_join(sampleMetadataAtomic, by = 'sample') %>%
+        #   .[[matchSampleColname]]
+
+        matchNames = mergeDataTable(colnames(edf)[2:ncol(edf)], sampleMetadataAtomicDT)[[matchSampleColname]]
         colnames(edf) = c('geneId', matchNames)
         edfListNow[[atomicStudyName]] = edf}
 
-      edfMerged = suppressWarnings(dplyr::bind_rows(edfListNow)) %>%
-        dplyr::group_by(geneId) %>%
-        dplyr::summarize_all(mergeFunc) %>%
-        data.frame(check.names=FALSE)
+      # edfMerged = suppressWarnings(dplyr::bind_rows(edfListNow)) %>%
+      #   dplyr::group_by(geneId) %>%
+      #   dplyr::summarize_all(mergeFunc) %>%
+      #   data.frame(check.names = FALSE)
+
+      edfMerged = data.frame(data.table(rbind(edfListNow)[[1]])[, lapply(.SD, mergeFunc)], by = geneId, check.names = FALSE)
```

Comment:
`by` is not an argument to `data.frame`, so this code should give an error. Have you tested this? I think the closing bracket is misplaced.

Reply:
I was testing each of the functions side by side with the dplyr ones, though its possible I mistakenly typed that or copied and pasted wrong when officially changing it. In any case, its been updated with the latest push.